code/modules/clothing/masks/gasmask.dm

```diff
@@ -31,6 +31,9 @@
 /obj/item/clothing/mask/gas/welding/attack_self(mob/user)
 	weldingvisortoggle(user)
 
+/obj/item/clothing/mask/gas/welding/up/equipped(mob/user, slot)
+	..()
+	weldingvisortoggle(user)	

```

Comment:
?

Reply:
Yeah that was a stupid way of achieving this. :slightly_frowning_face: 
Put it in initialize instead, like I should have.